File: data/src/main/java/com/joxad/zikobot/data/module/spotify_api/model/Item.java
package com.joxad.zikobot.data.module.spotify_api.model;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import org.parceler.Parcel;

import java.util.ArrayList;
import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@Parcel
@AllArgsConstructor(suppressConstructorProperties = true)
public class Item {

    @SerializedName("collaborative")
    @Expose
    public Boolean collaborative;
    @SerializedName("external_urls")
    @Expose
    public ExternalUrls externalUrls;
    @SerializedName("href")
    @Expose
    public String href;
    @SerializedName("id")
    @Expose
    public String id;
    @SerializedName("images")
    @Expose
    public List<Image> images = new ArrayList<Image>();
    @SerializedName("name")
    @Expose
    public String name;
    @SerializedName("owner")
    @Expose
    public Owner owner;
    @SerializedName("public")
    @Expose
    public Boolean _public;
    @SerializedName("snapshot_id")
    @Expose
    public String snapshotId;
    @SerializedName("tracks")
    @Expose
    public Tracks tracks;
    @SerializedName("type")
    @Expose
    public String type;
    @SerializedName("uri")
    @Expose
    public String uri;

    public Item() {
    }

}

File: data/src/main/java/com/joxad/zikobot/data/module/spotify_api/model/Item.kt
package com.joxad.zikobot.data.module.spotify_api.model

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName


import java.util.ArrayList

import lombok.AllArgsConstructor
import lombok.Data

class Item {

    @SerializedName("collaborative")
    @Expose
    var collaborative: Boolean? = null
    @SerializedName("external_urls")
    @Expose
    var externalUrls: ExternalUrls? = null
    @SerializedName("href")
    @Expose
    var href: String? = null
    @SerializedName("id")
    @Expose
    var id: String? = null
    @SerializedName("images")
    @Expose
    var images: List<Image> = ArrayList()
    @SerializedName("name")
    @Expose
    var name: String? = null
    @SerializedName("owner")
    @Expose
    var owner: Owner? = null
    @SerializedName("public")
    @Expose
    var _public: Boolean? = null
    @SerializedName("snapshot_id")
    @Expose
    var snapshotId: String? = null
    @SerializedName("tracks")
    @Expose
    var tracks: Tracks? = null
    @SerializedName("type")
    @Expose
    var type: String? = null
    @SerializedName("uri")
    @Expose
    var uri: String? = null


}
